cpp/tests/join/join_tests.cpp

```diff
@@ -1041,7 +1041,7 @@ TEST_F(JoinTest, EmptyRightTableInnerJoin)
   }
 
   {
-    cudf::hash_join hash_join(empty1, cudf::null_equality::EQUAL);
+    cudf::hash_join hash_join(empty1, true, cudf::null_equality::EQUAL);
```

Comment:
We should add an enum for this instead of a raw boolean type exposed to the public.

Reply:
Okay then I'll use `nullate::DYNAMIC`.

Reply:
Can we take out `nullate::DYNAMIC` from `column_device_view.cuh` and put it in another header? We use it quite a lot and it doesn't seem to have any device-specific implementations anyway. This doesn't have to be in this PR.

Reply:
I thought about that too. Where to put depends on discussion.

Reply:
Should be out of scope of this PR. I'll open another one for it.